### PR TITLE
fix(common): fallback to empty string for enums when validating (swc builder)

### PR DESCRIPTION
**Pull request by kamilmysliwiec**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #12680

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: packages/common/pipes/validation.pipe.ts [hunk 1]
```diff
@@ -208,8 +209,24 @@ export class ValidationPipe implements PipeTransform<any> {
...
+    if (
+      typeof metatype === 'function' ||
+      (metatype && 'prototype' in metatype && metatype.prototype?.constructor)
+    ) {
+      return {};
+    }
+    // Builder like SWC require empty string to be returned instead of an empty object
+    // when the value is nil and the metatype is not a class instance, but a plain object (enum, for example).
+    // Otherwise, the error will be thrown.
+    // @ref https://github.com/nestjs/nest/issues/12680
```
Comment (kamilmysliwiec) [inline packages/common/pipes/validation.pipe.ts:228]:
```suggestion
    // @see https://github.com/nestjs/nest/issues/12680
```

**Comment by JoCat** [comment]
Now there are no previous errors. But only the query gives 400 in the condition when the parameter is not specified, even though it is optional.
image
image

**Comment by kamilmysliwiec** [comment]
@JoCat could you create an issue with a minimum reproduction repository?

**Comment by JoCat** [comment]
There error is related and also reproduced in the same repository

**Comment by JoCat** [comment]
I tried to reproduce this problem again and I think I have figured out the cause. If you don't use ValidationPipe globally with `transform: true` specified, everything works correctly.

https://codesandbox.io/p/devbox/intelligent-zeh-mkyzmh

**Comment by lucas-gregoire** [comment]
I created a new issue for the 400 error now occurring: https://github.com/nestjs/nest/issues/14430

**Pull request merged**